### ISSUE TITLE
Crash at Worldgen with Promenade

### ISSUE THREAD

**Issue by Dragonfly-Totem** [quiqueck/BCLib#16]
### What happened?

Game loads without issue at this time, but when going to generate a new world game crashes back to MC Launcher.  Confirmed issue does not reoccur when BetterEnd 2.0.5 is removed; Issue does not seem related to BCLib 2.0.10 update.

### BCLib

2.0.10

### Fabric API

0.57.0

### Fabric Loader

0.14.8

### Minecraft

1.19

### Relevant log output

```shell
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 7/3/22, 2:27 PM
Description: mouseClicked event handler

java.lang.IllegalStateException: Trying to access unbound value 'ResourceKey[minecraft:worldgen/biome / promenade:dark_amaranth_forest]' from registry Registry[ResourceKey[minecraft:root / minecraft:worldgen/biome] (Stable)]
	at net.minecraft.class_6880$class_6883.comp_349(class_6880.java:159)
	at com.google.common.collect.Maps$9.transformEntry(Maps.java:2053)
	at com.google.common.collect.Maps$12.getValue(Maps.java:2101)
	at net.minecraft.class_5455.method_31141(class_5455.java:279)
	at net.minecraft.class_5455.method_40314(class_5455.java:266)
	at net.minecraft.class_525.method_41854(class_525.java:147)
	at net.minecraft.class_525.md2575d9$lambda$bcl_NewDefaultSettings$0$0(class_525.java:776)
	at net.minecraft.class_7237.method_42098(class_7237.java:24)
	at net.minecraft.class_525.method_31130(class_525.java:144)
	at net.minecraft.class_526.method_19944(class_526.java:72)
	at net.minecraft.class_4185.method_25306(class_4185.java:40)
	at net.minecraft.class_4264.method_25348(class_4264.java:16)
	at net.minecraft.class_339.method_25402(class_339.java:120)
	at net.minecraft.class_4069.method_25402(class_4069.java:27)
	at net.minecraft.class_312.method_1611(class_312.java:94)
	at net.minecraft.class_437.method_25412(class_437.java:491)
	at net.minecraft.class_312.method_1601(class_312.java:94)
	at net.minecraft.class_312.method_22686(class_312.java:165)
	at net.minecraft.class_1255.execute(class_1255.java:103)
	at net.minecraft.class_312.method_22684(class_312.java:165)
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:43)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3474)
	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:222)
	at net.minecraft.class_310.method_1523(class_310.java:1174)
	at net.minecraft.class_310.method_1514(class_310.java:734)
	at net.minecraft.client.main.Main.main(Main.java:237)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
```

### Other Mods

```shell
BetterEnd 2.0.5
Dawn API 3.5.0
Promenade 2.7.0
```

**Comment by quiqueck** [comment]
So, just to explain this a bit. The cores issue is, that Promenade is announcing new Biomes but does not immediately register them. There appear to be a lot of places that rely on announced biomes to be registered as well which is causing all those crashes. Not sure yet what is happening here tho...

**Comment by Dragonfly-Totem** [comment]
I can report it to the Dawn team if needed as well; I just figured here would probably be the best place to do first reporting.  I ain't a coder -- Just a general helpdesk grunt who can loosely follow troubleshooting rhythm.

**Comment by quiqueck** [comment]
No, that is not what I was trying to say :)
Just wanted to explain a bit why there are so many issues here.

**Comment by Dragonfly-Totem** [comment]
Ah, I dig it.  Thank you, and no worries!  Debugging never ends, man.  Thanks immensely for all the work and attention to the modstack.

**Comment by quiqueck** [comment]
Should be fixed with the just released BCLib 2.0.11.
